pkg/runtime/runtime.go

```diff
@@ -645,14 +645,22 @@ func (a *DaprRuntime) startHTTPServer(port, profilePort int, allowedOrigins stri
 }
 
 func (a *DaprRuntime) startGRPCInternalServer(api grpc.API, port int) error {
-	serverConf := grpc.NewServerConfig(a.runtimeConfig.ID, a.hostAddress, port, a.namespace, a.accessControlList.TrustDomain)
+	var trustDomain string
+	if a.accessControlList != nil {
+		trustDomain = a.accessControlList.TrustDomain
+	}
+	serverConf := grpc.NewServerConfig(a.runtimeConfig.ID, a.hostAddress, port, a.namespace, trustDomain)
 	server := grpc.NewInternalServer(api, serverConf, a.globalConfig.Spec.TracingSpec, a.globalConfig.Spec.MetricSpec, a.authenticator)
 	err := server.StartNonBlocking()
 	return err
 }
 
 func (a *DaprRuntime) startGRPCAPIServer(api grpc.API, port int) error {
-	serverConf := grpc.NewServerConfig(a.runtimeConfig.ID, a.hostAddress, port, a.namespace, a.accessControlList.TrustDomain)
+	var trustDomain string
+	if a.accessControlList != nil {
+		trustDomain = a.accessControlList.TrustDomain
+	}
+	serverConf := grpc.NewServerConfig(a.runtimeConfig.ID, a.hostAddress, port, a.namespace, trustDomain)
```

Comment:
we do not think we need this for gRPC API endpoint, do we?

Reply:
It is used for cert gen:-
	https://github.com/dapr/dapr/blob/ce89980eda9cc86efbc11a745e2959b1925236ca/pkg/grpc/server.go#L125

Reply:
I see. for service invoke api. thanks for the explanation.